docs/advanced-options.md

```diff
@@ -0,0 +1,134 @@
+- **OutputImageSize**: Performs image rescaling. The format is `wxh`, for example "640x480". (default - same as InputImageSize)
+- **OutputAspect**: Set the video display aspect ratio specified by aspect. For example "4:3", "16:9". (default - same as source)
+- **OutputVideoCodec**: Set the video codec. For example "mpeg4", "h264". (default - mpeg4)
+- **OutputVideoQuality**: Use fixed video quality level. Range is a integer number between 1 to 31, with 31 being the worst quality. (default - dynamically set by FFmpeg)
+
+You can also set default values for these options by adding additional attributes to the device resource **StartStreaming**.
+The attribute name consists of a prefix "default" and the option name.
+
+For example:
+```yaml
+deviceResources:
+  - name: "StartStreaming"
+   description: "Start streaming process."
+   attributes:
+     { command: "VIDEO_START_STREAMING",    
+       defaultInputFrameSize: "320x240", 
+       defaultOutputVideoQuality: "31" 
+     }
+   properties:
+     valueType: "Object"
+     readWrite: "W"
+```
+
+> NOTE: It's NOT recommended to set default video options in the [cmd/res/profiles/general.usb.camera.yaml](cmd/res/profiles/general.usb.camera.yaml) as they may not be supported by every camera.
+
+## Dynamic Discovery
+The device service supports [dynamic discovery](https://docs.edgexfoundry.org/2.1/microservices/device/Ch-DeviceServices/#dynamic-provisioning).
+During dynamic discovery, the device service scans all connected USB devices and sends the discovered cameras to Core Metadata.
+The device name of the camera discovered by the device service is comprised of Card Name and Serial Number, and the characters colon, space and dot will be replaced with underscores as they are invalid characters for device names in EdgeX.
+Take the camera Logitech C270 as an example, it's Card Name is "C270 HD WEBCAM" and the Serial Number is "B1CF0E50" hence the device name - "C270_HD_WEBCAM-B1CF0E50".
+
+> NOTE: Card Name and Serial number are used by the device service to uniquely identify a camera, although those cheaply mass-produced cameras may have the same serial number.
+
+### Enable the Dynamic Discovery function
+Dynamic discovery is disabled by default to save computing resources.
+If you want the device service to run the discovery periodically, enable it and set a desired interval.
+The interval value must be a [Go duration](https://pkg.go.dev/time#ParseDuration).
+
+[Option 1] Enable from the [configuration.toml](../cmd/res/configuration.toml)
+```yaml
+[Device] 
+...
+    [Device.Discovery]
+    Enabled = true
+    Interval = "1h"
+```
+
+[Option 2] Enable from the env
+```shell
+export DEVICE_DISCOVERY_ENABLED=true
+export DEVICE_DISCOVERY_INTERVAL=1h
+```
+
+To manually trigger a Dynamic Discovery, use this [device service API](https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/device-sdk/2.2.0#/default/post_discovery).
+
+### Provision watcher example
```

Comment:
Does the end-user need to execute this command? Unclear.... also what is the Provision Watcher and what is this example illustrating?